File: programs/enums/enum_exam.py
import enum

MALE = 1
FEMALE = 2

class Gender(enum.Enum):
    MALE = 1
    FEMALE = 2

for gender in Gender:
    print(gender.name, gender.value)

class Student:
    def __init__(self, name, gender):
        self.name = name
        self.gender = gender

kim = Student("Kim DukBae", Gender.MALE)

if kim.gender == Gender.MALE:
    print("Male student")
elif kim.gender == Gender.FEMALE:
    print("Female student")

if Gender.MALE == 1:
    print("Gender.MALE is 1")

# print(Gender.MALE + 1)  # ERROR

File: programs/enums/EnumExam.java
enum Gender {
    MALE(1), FEMALE(2);

    private final int value;

    Gender(int value) {
        this.value = value;
    }

    public int value() {
        return this.value;
    }
}

// enum Gender {
//     MALE, FEMALE;
// }

class Student {
    public String name;
    public Gender gender;

    Student(String name, Gender gender) {
        this.name = name;
        this.gender = gender;
    }
}

public class EnumExam {
    public static void main(String[] args) {
        Student kim = new Student("Kim DukBae", Gender.MALE);
        if(kim.gender == Gender.MALE) {
            System.out.println("Male Student");
        }
        if(kim.gender == Gender.FEMALE) {
            System.out.println("Female Student");
        }
        // kim.gender++;   // Error
        System.out.println(kim.gender.ordinal());
        // System.out.println(Gender.MALE == 0);   // Error
    }
}